test/TensorNetwork_test.jl

```diff
@@ -25,6 +25,83 @@
         end
     end
 
+    @testset "Base.similar" begin
+        listtensors = [Tensor(rand(2, 3, 4), (:i, :j, :k)), Tensor(rand(4, 3, 2), (:l, :j, :m))]
+        tn = TensorNetwork(listtensors)
+
+        similartn = similar(tn)
+
+        @test length(tensors(tn)) == length(tensors(similartn))
+        @test issetequal(inds(tn), inds(similartn))
+        @test all(map((tns,simtns)-> inds(tns)==inds(simtns), tensors(tn), tensors(similartn)))
```

Comment:
This is more readable:
```suggestion
        @test all(splat(==), zip(tensors(tn), tensors(similartn)))
```